src/index.ts

```diff
@@ -1,15 +1,102 @@
+import { DNSoverHTTPS } from "dohdec";
 import { getLogger } from "./util/logger";
 
-const { trace } = getLogger("sayHello");
+const { trace } = getLogger("index");
+
+const DoHResolver = new DNSoverHTTPS();
+
+type RecordTypes = "openatts";
+
+type BlockchainNetwork = "ethereum";
+
+type EthereumAddress = string;
+
+enum EthereumNetworkIds {
+  homestead = 1,
+  ropsten = 3
+}
+interface OpenAttestationsDNSTextRecord {
+  type: RecordTypes;
+  net: BlockchainNetwork;
+  netId: EthereumNetworkIds;
+  address: EthereumAddress;
+  dnssec: boolean;
+}
+
+/**
+ * Returns true for strings that are openattestation records
+ * @param txtDataString e.g: '"openatts net=ethereum netId=3 address=0x0c9d5E6C766030cc6f0f49951D275Ad0701F81EC"'
+ */
+const isOpenAttestationsRecord = (txtDataString: string) => {
+  return txtDataString.startsWith('"openatts');
+};
+
+/**
+ * Takes a string in the format of "key=value" and adds it to a JS object as key: value
+ * @param obj Object that will be modified
+ * @param keyValuePair
+ */
+const addKeyValuePairToObject = (obj: any, keyValuePair: string): any => {
+  const [key, value] = keyValuePair.split("=", 2);
```

Comment:
it's not what you expected => missing tests ?

Reply:
added tests and fixed